examples/automm/TCGA_cancer_survival/example_cancer_survival.py

```diff
@@ -0,0 +1,118 @@
+"""
+Example script to predict the vital status of patients with Head-Neck Squamous Cell Carcinoma.
+Dataset is originally from https://portal.gdc.cancer.gov/projects/TCGA-HNSC.
+Paper working on similar task: https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/s12859-019-2929-8.pdf
+"""
+
+import pandas as pd
+import numpy as np
+import argparse
+import os
+import random
+from autogluon.tabular import TabularPredictor, TabularDataset
+from autogluon.tabular.configs.hyperparameter_configs import get_hyperparameter_config
+import torch as th
+from sklearn.model_selection import train_test_split
+from autogluon.multimodal.utils import download
+import warnings
+warnings.filterwarnings('ignore')
+
+
+# Dataset information for TCGA dataset
+INFO = {
+    "name": "cancer_survival.tsv",
+    "url": "s3://automl-mm-bench/life-science/clinical.tsv",
+    "sha1sum": "6d19609c2a8492f767efd9f2c0b7687bcd3845a3"
+}
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description='The Basic Example of AutoGluon for TCGA dataset.')
+    parser.add_argument('--path', default='./dataset')
+    parser.add_argument('--test_size', default=0.3)
+    parser.add_argument('--shuffle', default=True)
+    parser.add_argument('--seed', type=int, default=123)
+    parser.add_argument('--task', choices=['TCGA_HNSC', 'adult'],
+                        default='adult')
+    parser.add_argument('--mode', choices=['FT_Transformer', 'all_models'],
+                        default='all_models')
+    parser.add_argument('--num_gpus', type=int, default=None)
+    parser.add_argument('--num_workers', type=int, default=12)
```

Comment:
Is it unable to infer the optimal value? Why default 12? Have you tried None/-1?

Reply:
There doesn't seems to be an easy of of setting the optimal value here. 'num_workers' seems to be relevant to [many factors](https://discuss.pytorch.org/t/guidelines-for-assigning-num-workers-to-dataloader/813/3). Here, 0 means single thread and -1 is not allowed. Should we just go with with the default setting with [MultiModal default parameter](https://github.com/awslabs/autogluon/blob/06fbc64cd1df53618b571c9a50835385d7b28c58/multimodal/src/autogluon/multimodal/configs/environment/default.yaml#:~:text=env%3A,%22ddp_spawn%22)? In that case num_gpus=-1 and num_workers=2.

Reply:
Default value makes sense.